src/facade/dragonfly_connection.cc

```diff
@@ -1445,7 +1445,10 @@ auto Connection::IoLoop() -> variant<error_code, ParserStatus> {
           });
         }
 
-        DCHECK_GT(io_buf_.AppendLen(), 0U);
+        if (io_buf_.AppendLen() == 0U) {
```

Comment:
please add a comment stating that it can happen with memcached but does not with RedisParser, 
because RedisParser fully consumes the passed buffer